src/components/tasks/change-proposals-drawer.tsx

```diff
@@ -0,0 +1,259 @@
+import { Box, Button, Drawer, LinearProgress, List, ListItem, Typography } from "@mui/material";
+import { useEffect, useRef, useState } from "react";
+import { useTranslation } from "react-i18next";
+import RemoveRedEyeIcon from "@mui/icons-material/RemoveRedEye";
+import DoneIcon from "@mui/icons-material/Done";
+import DeleteOutlineIcon from "@mui/icons-material/DeleteOutline";
+import { ChangeProposal, ChangeProposalStatus, Task } from "generated/client";
+import { DateTime } from "luxon";
+import { useFindUsersQuery } from "hooks/api-queries";
+import DragHandleIcon from "@mui/icons-material/DragHandle";
+
+/**
+ * Change proposals component properties
+ */
+interface Props {
+  changeProposals?: ChangeProposal[];
+  tasks?: Task[];
+  selectedChangeProposalId: string;
+  setSelectedChangeProposalId: (selectedChangeProposalId: string) => void;
+  loading: boolean;
+  updateChangeProposalStatus: (
+    changeProposalId: string,
+    changeProposal: ChangeProposal,
+    status: ChangeProposalStatus,
+  ) => void;
+}
+
+/**
+ * Change proposals drawer component
+ *
+ * @props props component properties
+ */
+const ChangeProposalsDrawer = ({
+  changeProposals,
+  tasks,
+  selectedChangeProposalId,
+  setSelectedChangeProposalId,
+  loading,
+  updateChangeProposalStatus,
+}: Props) => {
+  const { t } = useTranslation();
+  const [open, setOpen] = useState(false);
+  const [height, setHeight] = useState(150);
+  const [maxHeight, setMaxHeight] = useState<number | null>(null);
+  const startY = useRef(0);
+  const startHeight = useRef(height);
+  const contentRef = useRef<HTMLDivElement>(null);
+
+  const proposalCreatorUsersIds = [
+    ...new Set(
+      changeProposals
+        ?.flatMap((proposal) => proposal.metadata?.creatorId)
+        .filter((creatorId): creatorId is string => creatorId !== undefined),
+    ),
+  ];
```

Comment:
Should we have it wrapped with useMemo to avoid recalculation on every re-render?